Cython/Compiler/Nodes.py

```diff
@@ -559,6 +559,14 @@ def analyse(self, base_type, env, nonempty=0, visibility=None, in_pxd=False):
         return self.base.analyse(ref_type, env, nonempty=nonempty, visibility=visibility, in_pxd=in_pxd)
 
 
+class CppRvalueReferenceDeclaratorNode(CReferenceDeclaratorNode):
```

Comment:
I'm using inheritance here purely for code reuse. An rvalue reference does not have an "is-a" relationship with lvalue reference. Open to other suggestions here, and ditto with  CppRvalueReferenceType/CReferenceType.

Reply:
A common base class sounds good. I'll go with that

Reply:
Done.